sphinx/ext/autodoc/__init__.py

```diff
@@ -1370,7 +1386,36 @@ def format_signature(self, **kwargs: Any) -> str:
         if self.doc_as_attr:
             return ''
 
-        return super().format_signature(**kwargs)
+        sig = super().format_signature(**kwargs)
+        sigs = []
+        overloaded = False
+        fullqualname = None
+        if self._signature_method and self.analyzer:
+            if self._signature_method == '__call__':
+                metaclass = type(self.object)
+                fullqualname = '.'.join([metaclass.__qualname__, '__call__'])
+            else:
+                fullqualname = '.'.join(self.objpath + [self._signature_method])
```

Comment:
Would it be better to store `signature_qualname` instead of `signature_method`? Then you wouldn't have this metaclass logic in two places.